Support_Your_Locals/Controllers/AdminController.cs

```diff
@@ -15,38 +14,50 @@ namespace Support_Your_Locals.Controllers
     public class AdminController : Controller
     {
 
-        private IServiceRepository repository;
         private IConfiguration configuration;
-        public AdminController(IServiceRepository repo, IConfiguration config)
+        private ILegacyServiceRepository repository;
+
+        public AdminController(ILegacyServiceRepository repo, IConfiguration config)
         {
-            repository = repo;
             configuration = config;
+            repository = repo;
+
         }
         [HttpGet]
         [Authorize(Roles = "Administrator")]
         public ActionResult Index()
         {
+            var businesses = repository.GetBusinesses();
+            var usersBusinesses = businesses.GroupBy(business => business.User);
+            var users = repository.GetUsers().Where(notAdmin);
```

Comment:
From where is this not admin coming? Are admins in the same schema as standart users?

Reply:
Probably in future there will be added roles, so if some kind of person will have "Administrator" role in cookie then he will have access to this page

Reply:
We already have roles in our authentication mechanism. That's why there is `[Authorize(Roles = "Administrator")]` so only users with admin role can access this method. 
`isAdmin` is predicate function to filter out users that are not admins from user list since they are stored in the same schema. I think admin users are not relevant in the admin page that is why I filter them out

Reply:
That explains it. Thanks.